api/v1beta1/register.go

```diff
@@ -35,6 +36,13 @@ func addKnownTypes(scheme *runtime.Scheme) error {
 		&DisruptionCronList{},
 	)
 
+	scheme.AddKnownTypes(schema.GroupVersion{Group: GroupName, Version: runtime.APIVersionInternal},
```

Comment:
I don't think we need this.

Reply:
It is needed for the `Watch` method to work. The watcher needs `Disruption` kind to be registered in the internal version of the group in the scheme, otherwise it throws an error. I can move into `install.go` instead maybe?

Reply:
This seems like the right place to put this. But how is this different than the call to `scheme.AddKnownTypes` above? Do we know why we end up needing to add it to two versions?

Reply:
The `clientset` generation process is dependent on the structure of the folders in the project. In this case, it has generated a `clientset` for what it interprets as the internal version, rather than for `v1beta1`. As a result, when we use the `clientset` to interact with disruptions, it attempts to manipulate the internal version. However, it can't find the resource definition because it doesn't exist. 
If we prefer to have it in a more standard path, it would need substantial modifications to the controller.

Reply:
The constant [GroupVersion](https://github.com/DataDog/chaos-controller/blob/main/api/v1beta1/groupversion_info.go#L30) used line 32 is the same as the one defined [line 39](https://github.com/DataDog/chaos-controller/pull/860/files#diff-ed6c59a090681be5bf9f8de4020890ef2433aefb49a758a508383d94f5adda90R39).
So it looks duplicated. Does the unit tests are working if you remove this block of code?

Reply:
The GroupVersion on line 32 is `schema.GroupVersion{Group: 'chaos.datadoghq.com', Version: 'v1beta1'}`, the GroupVersion on line 39 is  `schema.GroupVersion{Group: 'chaos.datadoghq.com', Version: '__internal'}`. If I remove block of code between lines 39 and 44, the tests won't work. 

Here is an error message that appears when I remove that block of code:
``` go
Status:Failure,Message:an error on the server (\"unable to decode an event from the watch stream: unable to decode watch event: no kind \\\"Disruption\\\" is registered for the internal version of group "chaos.datadoghq.com\\\" in scheme \\\"/Users/diyara.bissembayeva/chaos-controller/clientset/v1beta1/scheme/register.go:17\\\"\") has prevented the request from succeeding
```
This just goes back to my comment above.

Reply:
Hmm, are we doing something wrong elsewhere then? The Disruption Kind definitely should only exist in the v1beta1 version

Reply:
we use the [client-gen](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/generating-clientset.md) tool to generate our `clientset`, which relies on specific folder structures and tagging conventions to determine versions. According to the `client-gen` docs, the folder struscture is expects to find API types follows the format `pkg/apis/${GROUP}/${VERSION}/types.go`.

Currently, our project's folder structure led `client-gen` to misinterpret our main version as `internal`, rather than the intended `v1beta1`. This results in a `clientset` that tries to interact with a non-existent internal version, causing the issues we are seeing.

To address this, we would need to reorganize of the chaos-controller's folder structure. I think this is a substantial modification which falls outside the scope of the current PR.

Reply:
Agreed that we wouldn't want to re-organize in this PR. But does the new clientset still let us query v1beta1? (seems like it?) Are we able to force specifying a v1beta1 version in CLI args, rather than inferring from the package structure? But fair, we can discuss this separately from this PR, imo

Reply:
Indeed, you right @diyarab the version is not the same and knows I understand the problem. 

Thanks for explanations 🙏  And I agree we should tackle this in another PR to have a correct usage of the schema.